src/main/java/Task.java

```diff
@@ -0,0 +1,44 @@
+public abstract class Task {
+    private String name;
+    private boolean isDone;
+
+    public Task(String name) {
+        this.name = name;
+        this.isDone = false;
+    }
+
+    public Task(String name, boolean isDone) {
+        this.name = name;
+        this.isDone = isDone;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setDone(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    public boolean hasDone() {
```

Comment:
Following the coding style we are given, maybe isDone would be a better name for this method.